### PR TITLE
Backport podspec and binary building to v0.4.1

### PR DESCRIPTION
**Pull request by alloy**
## Summary

Backport podspec and binary building to v0.4.1. Note that this reverts the 0.4.2-rc1 changes!

## Test Plan

See [steps to create a NPM package](https://github.com/facebook/hermes/blob/45177faae0252da470ca49d67499b4c913806adf/npm/hermes-engine-darwin/BUILD.md) and use it with [this RN macOS change](https://github.com/facebook/react-native/commit/d4d07b96d62e2d40b24bd908551850772855f2a6).

### CODE REVIEW

**Comment by alloy** [comment]
@willholen Ok, this should be good to go; I’ve tested the build artefact and [the steps to create the NPM package](https://github.com/facebook/hermes/blob/45177faae0252da470ca49d67499b4c913806adf/npm/hermes-engine-darwin/BUILD.md) locally with success. Can you publish the release?

**Comment by willholen** [comment]
Could you update this with the same changes from https://github.com/facebook/hermes/pull/297/files/3e4904476bc65de821c7bb43d602a8da8e675b36 ? (there's a few "Show resolved" buttons to see what exactly they were)

**Comment by alloy** [comment]
@willholen Good call 👍 Done.

**Comment by Huxpro** [comment]
Hey @alloy, thanks for the work!

Since we already have a 0.4.1 section in https://github.com/facebook/hermes/releases, I think we can make an 0.4.3 exclusively for macOS. Could you change the version in the code (`package.json`, `CMakefile`, `.gradle`, `podspec`) all to the 0.4.3?

The plan is that we can only `npm publish` the `hermes-engine-darwin@0.4.3` and not the others.

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,18 +22,25 @@ if (POLICY CMP0075)
...
+  cmake_policy(SET CMP0026 OLD)
+endif()
+
+# This must be consistent with the release_version in:
+# - android/build.gradle
+# - npm/package.json
+# - hermes.podspec
 project(Hermes
-        VERSION 0.4.2
+        VERSION 0.4.1
```
Comment (Huxpro) [inline CMakeLists.txt:37]: 0.4.3

File: hermes.podspec [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def self.build_dir
+    BUILD_TYPE == :release ? "build_release" : "build"
+  end
+end
+
+Pod::Spec.new do |spec|
+  spec.name        = "hermes"
+  spec.version     = "0.4.1"
```
Comment (Huxpro) [inline hermes.podspec:33]: 0.4.3

File: npm/hermes-engine-darwin/package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "hermes-engine-darwin",
+  "version": "0.4.1",
```
Comment (Huxpro) [inline npm/hermes-engine-darwin/package.json:3]: 0.4.3

File: npm/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hermes-engine",
-  "version": "0.4.2-rc1",
+  "version": "0.4.1",
```
Comment (Huxpro) [inline npm/package.json:3]: 0.4.3

File: android/build.gradle [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 // This must be consistent with the release_version in npm/package.json
 // and the HERMES_RELEASE_VERSION in CMakeLists.txt
-def release_version = "0.4.2-rc1"
+def release_version = "0.4.1"
```
Comment (Huxpro) [inline android/build.gradle:10]: 0.4.3

**Comment by alloy** [comment]
@Huxpro Is it ok to _not_ put the proxy code back that was added in the 0.4.2 RC, so it otherwise remains equal to the existing 0.4.1 release?

**Comment by Huxpro** [comment]
> @Huxpro Is it ok to _not_ put the proxy code back that was added in the 0.4.2 RC, so it otherwise remains equal to the existing 0.4.1 release?

Yes! The plan is to only turn on Proxy at RC releases (for now).

**Comment by Huxpro** [comment]
Hey @alloy, since the PR looks otherwise great. I think we can just merge it and I will take care of the version bump thing. Thanks for the work!

**Comment by alloy** [comment]
@Huxpro Awesome, thanks so much! 🙏

**Comment by alloy** [comment]
@Huxpro @willholen Could you give me a ping when you’ve published the macOS version of the package?

**Pull request merged**